R/built.R

```diff
@@ -1,8 +1,9 @@
 
 idesc_get_built <- function(self, private) {
-  built <- unname(self$get("Built"))
+  built <- gsub("\\n", "", unname(self$get("Built")))
   if (is.na(built)) stop("No ", sQuote('Built'), " field found")
-  built <- as.list(strsplit(built, "; ")[[1L]])
+  built <- lapply(as.list(strsplit(built, "; ")[[1L]]), gsub,
```

Comment:
What do we need `lapply()` here for if we didn't need it before? I _think_ we still should have `length(built) == 1` even for multi-line fields?

Can we instead use `strsplit(built, "^[[:space:]]*;[[:space:]]+")[[1L]]` ?

Reply:
Splitting on leading and trailing space does not produce a list of length 4. Also I just now realized `stringr::str_trim` is already used in this package, so I replaced `gsub` in the `lapply`. Leading whitespace is introduced because of the new line. I'll commit that change if this implementation is okay with you. Reprex below:

```
library(devtools)
load_all()
self <- description$new(test_path("D16"))

built <- gsub("\\n", "", unname(self$get("Built")))
if (is.na(built)) stop("No ", sQuote('Built'), " field found")
built <- lapply(as.list(strsplit(built, "; ")[[1L]]), str_trim)
if (length(built) != 4L) {
  stop(sQuote('Built'), " field is corrupted")
}
names(built) <- c("R", "Platform", "Date", "OStype")
built[["R"]] <- R_system_version(sub("^R ([0-9.]+)", "\\1", built[["R"]]))
built

# $R
# [1] ‘3.4.1’
# 
# $Platform
# [1] "x86_64-apple-darwin15.6.0"
# 
# $Date
# [1] "2017-09-14 20:30:19 UTC"
# 
# $OStype
# [1] "unix"

built <- gsub("\\n", "", unname(self$get("Built")))
if (is.na(built)) stop("No ", sQuote('Built'), " field found")
built <- as.list(strsplit(built, "^[[:space:]]*;[[:space:]]+")[[1L]])
if (length(built) != 4L) {
  stop(sQuote('Built'), " field is corrupted")
}

# Error: ‘Built’ field is corrupted
```